tests/test_api.py

```diff
@@ -0,0 +1,860 @@
+
+        ecosystems = Ecosystem.objects.all()
+        self.assertEqual(len(ecosystems), 2)
+
+        eco1 = ecosystems[0]
+        self.assertEqual(eco1.id, self.eco_bitergia.id)
+        self.assertEqual(eco1.name, 'Bitergia')
+
+        eco2 = ecosystems[1]
+        self.assertEqual(eco2.id, self.eco_libresoft.id)
+        self.assertEqual(eco2.name, 'Libresoft')
+
+    def test_delete_non_existing_ecosystem(self):
+        """Check if it fails when deleting a non existing ecosystem"""
+
+        trx_date = datetime_utcnow()  # After this datetime no transactions should be created
+
+        with self.assertRaisesRegex(NotFoundError, ECOSYSTEM_NOT_FOUND_ERROR.format(eco_id='11111111')):
+            api.delete_ecosystem(self.ctx, 11111111)
+
+        # Check if there are no transactions created when there is an error
+        transactions = Transaction.objects.filter(created_at__gt=trx_date)
+        self.assertEqual(len(transactions), 0)
+
+    def test_ecosystem_id_none(self):
+        """Check if it fails when ecosystem id is `None`"""
+
+        trx_date = datetime_utcnow()  # After this datetime no transactions should be created
+
+        with self.assertRaisesRegex(InvalidValueError, ECOSYSTEM_ID_NONE_OR_EMPTY_ERROR):
+            api.delete_ecosystem(self.ctx, ecosystem_id=None)
+
+        # Check if there are no transactions created when there is an error
+        transactions = Transaction.objects.filter(created_at__gt=trx_date)
+        self.assertEqual(len(transactions), 0)
+
+    def test_ecosystem_id_empty(self):
+        """Check if it fails when ecosystem id is empty"""
+
+        trx_date = datetime_utcnow()  # After this datetime no transactions should be created
+
+        with self.assertRaisesRegex(InvalidValueError, ECOSYSTEM_ID_NONE_OR_EMPTY_ERROR):
+            api.delete_ecosystem(self.ctx, ecosystem_id='')
+
+        # Check if there are no transactions created when there is an error
+        transactions = Transaction.objects.filter(created_at__gt=trx_date)
+        self.assertEqual(len(transactions), 0)
+
+    def test_ecosystem_id_whitespaces(self):
+        """Check if it fails when ecosystem id is composed by whitespaces"""
+
+        trx_date = datetime_utcnow()  # After this datetime no transactions should be created
+
+        with self.assertRaisesRegex(InvalidValueError, ECOSYSTEM_ID_INVALID_LITERAL):
+            api.delete_ecosystem(self.ctx, ecosystem_id='   ')
+
+        with self.assertRaisesRegex(InvalidValueError, ECOSYSTEM_ID_INVALID_LITERAL):
+            api.delete_ecosystem(self.ctx, ecosystem_id='\t')
+
+        with self.assertRaisesRegex(InvalidValueError, ECOSYSTEM_ID_INVALID_LITERAL):
+            api.delete_ecosystem(self.ctx, ecosystem_id=' \t  ')
+
+        # Check if there are no transactions created when there is an error
+        transactions = Transaction.objects.filter(created_at__gt=trx_date)
+        self.assertEqual(len(transactions), 0)
```

Comment:
I don't think these are a valid test for the operation. The operation needs an `id` not a string.

Reply:
I renamed this test. The initial idea was to test an "invalid literal" error, because I realized the value in this field is going to be converted (at least it tries to convert it). For instance, if you set the id '3' (string), it will cast it to the integer 3, so the idea was to cause this error.

Reply:
The method shouldn't convert anything. If the method needs an integer it should only work with an integer to avoid further problems.

Reply:
Ok, copy that! What I meant is about Django's behaviour when looking for this ID field (`AutoField`). In `find_ecosystem` method, when performing a query to get Ecosystem objects using their id (`ecosystem = Ecosystem.objects.get(id=ecosystem_id)`), Django makes an automatic cast of the value of `ecosystem_id`, as `AutoField` does not limit its id values to integers. I'm including an additional check so non-integer values are not allowed.

Reply:
When this ID parameter is set in the query, even if it is set as an integer the mutation takes this argument always as an integer. That's why I decided to force the conversion of the scalar value before calling the corresponding API method.